src/android/SimpleCameraPreview.java

```diff
@@ -128,51 +168,98 @@ public void onProviderDisabled(String provider) {
+                        permissionsGranted = false;
+                        break;
+                    }
+                }
+            }
 
-        if (requestCode == GEO_REQ_CODE)
-            fetchLocation();
+            if (!permissionsGranted) {
+                boolean permissionAlwaysDenied = false;
+
+                for (String permission : permissions) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(cordova.getActivity(), permission)) {
+                        cordova.requestPermissions(this, REQUEST_CODE_PERMISSIONS, REQUIRED_PERMISSIONS);
+                    } else {
+                        if (ActivityCompat.checkSelfPermission(cordova.getContext(), permission) == PackageManager.PERMISSION_DENIED) {
+                            permissionAlwaysDenied = true;
+                        }
+                    }
+                }
+
+                if (permissionAlwaysDenied) {
+                    AlertDialog.Builder builder = new AlertDialog.Builder(cordova.getContext());
+                    builder.setTitle("Permissions required")
+                            .setMessage("Go to app info > permissions and allow them")
+                            .setCancelable(false)
+                            .setPositiveButton("App info", ((dialogInterface, i) -> {
+                                Intent intent = new Intent(
+                                        Settings.ACTION_APPLICATION_DETAILS_SETTINGS,
+                                        Uri.fromParts("package", cordova.getActivity().getPackageName(), null)
+                                );
+                                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                                cordova.getActivity().startActivity(intent);
+                                cordova.getActivity().finish();
+                            }))
+                            .setNegativeButton("Cancel", ((dialogInterface, i) -> {
+                                cordova.getActivity().finish();
+                            }))
+                            .create()
+                            .show();
+                }
+            } else {
+                enable(this.options, this.callbackContext);
+                fetchLocation();
+            }
+        }
     }
 
     public void fetchLocation() {
-        if (ContextCompat.checkSelfPermission(cordova.getActivity(), android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-            if (locationManager == null)
+        if (ContextCompat.checkSelfPermission(cordova.getActivity(), REQUIRED_PERMISSIONS[1]) == PackageManager.PERMISSION_GRANTED) {
+            if (locationManager == null) {
                 locationManager = (LocationManager) cordova.getActivity().getSystemService(Context.LOCATION_SERVICE);
+            }
             Location cachedLocation = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
```

Comment:
Is this the FINE_LOCATION ? From https://developer.android.com/reference/android/location/LocationManager#NETWORK_PROVIDER, 
> Operation of this provider may require a data connection.
The app should be usable completely offline.

Reply:
yes